src/libs/SidebarUtils.ts

```diff
@@ -379,15 +390,15 @@ function getReasonAndReportActionThatHasRedBrickRoad(
     const transactionThreadReportID = getOneTransactionThreadReportID(report.reportID, reportActions ?? []);
     if (transactionThreadReportID) {
         const transactionID = getTransactionID(transactionThreadReportID);
-        const transaction = getTransaction(transactionID);
+        const transaction = allTransactions?.[`${ONYXKEYS.COLLECTION.TRANSACTION}${transactionID}`];
```

Comment:
similarly to report attributes, I think we should get rid of the allTransactions and similar from the utils files as if the transaction changes the component would not re-render